driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -16,10 +16,77 @@
 
 package com.mongodb.client.model.expressions;
 
+import org.bson.conversions.Bson;
+
+import java.time.Instant;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+
 /**
  * Expresses a document value. A document is an ordered set of fields, where the
  * key is a string value, mapping to a value of any other expression type.
  */
 public interface DocumentExpression extends Expression {
 
+    DocumentExpression setField(String path, Expression exp);
+
+    DocumentExpression unsetField(String path);
+
+
+    BooleanExpression getBoolean(String field);
```

Comment:
Let's also support `Expression get(String field)` for cases where you don't know the type and either 

1. don't care what it is, you just want to include it in a projection or do only Expression-y things with it.
2. want to do a type test on it, e.g. `doc.get("unknown").isBooleanOr(of(true))`

Reply:
Added `getField`.